projects/views.py

```diff
@@ -2,83 +2,35 @@
 from rest_framework.response import Response
-from rest_framework.views import APIView
 
 from components.filters import ComponentFilter
 from components.models import Component
 from components.serializers import ComponentListBasicSerializer
 
-from .models import Project
-from .serializers import (
-    ProjectControlSerializer,
-    ProjectListSerializer,
-    ProjectSerializer,
-)
+from blueprintapi.filters import ObjectPermissionsFilter
+from projects.models import Project
+from projects.serializers import ProjectControlSerializer, ProjectListSerializer, ProjectSerializer
 
 
 class ProjectsListViews(generics.ListCreateAPIView):
-
     queryset = Project.objects.all().order_by("pk")
     serializer_class = ProjectListSerializer
+    filter_backends = [ObjectPermissionsFilter, ]
 
-    def list(self, request, *args, **kwargs):
-        projects = self.get_queryset()
-        serializer = ProjectListSerializer(projects, many=True)
-        return Response(serializer.data, status=status.HTTP_200_OK)
-
-
-class ProjectsDetailView(APIView):
-    def get_object(self, project_id):
-        try:
-            return Project.objects.get(id=project_id)
-        except Project.DoesNotExist:
-            return None
 
-    def get(self, request, project_id, *args, **kwargs):
-        project_instance = get_object_or_404(Project, pk=project_id)
-        serializer = ProjectSerializer(project_instance)
-        return Response(serializer.data, status=status.HTTP_200_OK)
+class ProjectsDetailView(generics.RetrieveUpdateDestroyAPIView):
+    queryset = Project.objects.all()
+    serializer_class = ProjectSerializer
+    lookup_url_kwarg = "project_id"
 
-    def patch(self, request, project_id, *args, **kwargs):
-        project_instance = self.get_object(project_id)
-        if not project_instance:
-            return Response(
-                {"response": "The project you are looking for does not exist"},
-                status=status.HTTP_400_BAD_REQUEST,
-            )
-        serializer = ProjectSerializer(
-            instance=project_instance, data=request.data, partial=True
-        )
-        if serializer.is_valid():
-            serializer.save()
-            return Response(serializer.data, status=status.HTTP_200_OK)
-        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
-
-    def delete(self, request, project_id, *args, **kwargs):
-        project_instance = self.get_object(project_id)
-        if not project_instance:
-            return Response(
-                {"response": "The project you are looking for does not exist"},
-                status=status.HTTP_400_BAD_REQUEST,
-            )
-        project_instance.delete()
-        return Response(
-            {"response": "You have succesfully deleted the project!"},
-            status=status.HTTP_200_OK,
-        )
 
+class ProjectAddComponentView(generics.GenericAPIView):
+    queryset = Project.objects.all()
+    lookup_field = "pk"
```

Comment:
Need to actually use this in the `post` method. I forgot to add it in.

Reply:
Just kidding, this would require updates to the url to utilize. Need to address in a later PR.